### PR TITLE
Add WarriorJS to Docusaurus users

### PR DESCRIPTION
**Pull request by olistic**
## Motivation

[WarriorJS](https://warrior.js.org) is using Docusaurus for its documentation, added it to the users.

Also, the users array was growing quite large in `siteConfig.js`, so I took the liberty to extract it into a file inside of the `website/data` directory (also new). I'm fine with reverting this change if you don't think is a good one.

The one thing I'm now sure about is the changes to `package-lock.json` that were done when I run `npm install` on the root of the project. I don't know exactly why installing the dependencies would rewrite the lock file, but I suspect it could have something to do with the version of npm I'm using. Just in case:

```sh
$ node --version
v8.11.1
$ npm --version
6.0.0
```

Please let me know if I have to do something with regards to that. And thanks!

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

**Same users being shown in the landing page:**
screen shot 2018-05-09 at 14 28 43

**WarriorJS now added to the users page:**
screen shot 2018-05-09 at 14 29 01

## Related PRs

N/A

### CODE REVIEW

File: website/siteConfig.js [hunk 1]
```diff
@@ -5,156 +5,10 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-// List of projects/orgs using your project for the users page.
```
Comment (yangshun) [inline website/siteConfig.js]: Thanks for the fix! Let's save it as `users.json` instead so that we don't need an extra yaml loader.

**Comment by olistic** [comment]
@yangshun Done! Changed it to JSON, and also reverted the changes to `package-lock.json`.

**Comment by JoelMarcey** [comment]
Your project looks fun, @olistic 😄

**Comment by olistic** [comment]
Hey @JoelMarcey! Thanks 😄 It's still in its early stages, I'm focusing now on building a community of players and contributors. If you guys know people that could be interested I'd really appreciate help spreading the word out!

And thanks to you both and the rest of the people involved in Docusaurus 🙏 It helped me bring my documentation site to life with little to no effort and just worry about the content and nothing else.

cc. @yangshun

**Comment by yangshun** [comment]
Very happy to hear that! 😄

**Comment by JoelMarcey** [comment]
@olistic I believe this is the first Docusaurus site for a game, so I think I tweeted about it from our Docusaurus twitter account https://twitter.com/docusaurus/status/994333018071298048 😄

**Comment by olistic** [comment]
That's awesome @JoelMarcey! Thanks!

**Pull request merged**